app/lib/features/segment_deletion_config.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Features
+  class SegmentDeletionConfig
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(config = {})
+      format_config(config)
+      check_validity(config)
+    end
+
+    def self.valid_environment?
+      # This is because this feature destroys real data unless the call to segment is stubbed and without real credentials in the config (test environment),
+      # so we prevent this feature to be used in preview or development thinking that t be just checking around while actually destroying real data by accident
+      env = Rails.env
+      env.production? || env.test?
+    end
+
+    def self.log_disabled_warning
+      warn_message = '[WARNING] Can\'t enable "deletion of users in segment.'
+      warn_message = if valid_environment?
+                       "#{warn_message} Please revise your config."
+                     else
+                       "#{warn_message} The #{Rails.env} environment is not valid for this feature."
+                     end
+      Rails.logger.warn warn_message
+    end
+
+    attr_reader :config, :enabled
+    alias enabled? enabled
+
+    private
+
+    def format_config(config)
+      config = (config || {}).symbolize_keys
+      @config = ActiveSupport::OrderedOptions.new.merge(config)
+    end
+
+    def check_validity(initial_config)
+      valid_config = %i[email password].all? { |key| config.key?(key) }
+      my_class = self.class
+      return if (@enabled = valid_config && my_class.valid_environment?) || initial_config.blank?
```

Comment:
It looks like `return unless initial_config.present?` could be the first instruction of the method.